Expensify/App
# Surface room name errors and warn users about reserved policy room name error

CC: @jasperhuangg @luacmartins 

### Details
Warns users not to use reserved names for rooms. Also adds an inline error so that users can see what the error they're seeing is.

### Fixed Issues
Follow up to: https://github.com/Expensify/Auth/pull/6291

### Tests/ QA Steps
1. Go to the green plus on the bottom left and click on New Room
2. Verify like in the screenshot below, if you try to use `#admins` or `#announce` for the room name, it won't let you.
3. Also verify that it shows you the right error message (see screenshot).
4. Verify that you also see an error if you try to use the same name as an existing policy room in that same workspace.


### Tested On


### Screenshots


#### Web
![image](https://user-images.githubusercontent.com/19364431/149554203-ece9fc1f-0321-4078-9ff9-fe872ed70b93.png)
![image](https://user-images.githubusercontent.com/19364431/149554727-e7371380-694e-4369-9108-16ec384a522f.png)
